### PR TITLE
[wip] Add pyobjc to nixpkgs

### PR DESCRIPTION
**Pull request by matthewbauer**
###### Motivation for this change

This adds the "pyobjc" python module to nixpkgs. It is only available for OS X, however. Some of the frameworks failed with 10.9 SDK, so they have been disabled.

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/pyobjc/default.nix [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+      meta = {
+        platforms = stdenv.lib.platforms.darwin;
+        description = "Wrappers for the framework ${_name} on Mac OS X";
+        maintainers = [ stdenv.lib.maintainers.matthewbauer ];
+        license = stdenv.lib.licenses.mit;
+      };
+    };
+
+  frameworks = rec { # 3.1.1 corresponds to SDK 10.9 because we only have CF 10.9
+    Accounts = buildPyObjcFramework "Accounts" "3.1.1" "1jgasyz88mhl58ks1k06qqxd6ifzbkcjkir6ny0hjr6vc07lk99h" [ Cocoa ];
```
Comment (FRidh) [inline pkgs/development/python-modules/pyobjc/default.nix]: How do you deal with updates of these packages? Do you edit these hashes manually or automatically?

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -14496,6 +14529,25 @@ in {
...
 
+  modulegraph = buildPythonPackage rec {
+    name = "modulegraph-0.13";
+
+    src = pkgs.fetchurl {
+      url = "mirror://pypi/m/modulegraph/${name}.tar.gz";
+      sha256 = "1fp58318n7rabxakis5cw4z2lcyimsp167hlfl6y0mvf7zx8dxai";
+    };
+
+    doCheck = false;
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: can you add a reason why you disable tests

File: pkgs/top-level/python-packages.nix [hunk 3]
```diff
@@ -19375,6 +19427,27 @@ in {
...
 
+  py2app = buildPythonPackage rec {
+    name = "py2app-0.10";
+
+    src = pkgs.fetchurl {
+      url = "mirror://pypi/p/py2app/${name}.tar.gz";
+      sha256 = "0pmaxrvx2vnlgsdlbzqr32v2j0n81n07x4xrc2nigfdnrxsxvcc2";
+    };
+
+    doCheck = false;
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: here as well

**Review by FRidh — CHANGES_REQUESTED** [review]
Could you put the new Python packages as `<name>/default.nix` in `python-modules`.

**Comment by FRidh** [comment]
status update?

**Comment by matthewbauer** [comment]
I've gotten stuck on it and the reason I needed pyobjc (for Transmission) is not really necessary now that we have the GTK backend working on macOS. It was never working correctly anyway. Anybody can work off of this if they will need it.

**Pull request closed without merging**